create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -86,6 +95,45 @@ private Reference generateReference(DocTp docTp) {
         return reference;
     }
 
+    private List<TitletextTp> getTitleText(DocTp docTp) {
+        return docTp.getItem().getItem().getBibrecord().getHead().getCitationTitle().getTitletext();
+    }
+
+    private String extractMainTitle(DocTp docTp) {
+        StringBuilder mainTitleBuilder = new StringBuilder();
+        getTitleText(docTp)
+            .stream()
+            .filter(titleTextTp -> titleTextTp.getOriginal().equals(YesnoAtt.Y))
+            .findFirst()
+            .ifPresent(part -> mainTitleBuilder.append(getContentWithXmlTags(part.getContent())));
+        return mainTitleBuilder.toString();
+    }
+
+    private String getContentWithXmlTags(List<Object> contents) {
+        StringBuilder contentStringBuilder = new StringBuilder();
+        contents.forEach(content -> {
+            if (content instanceof String) {
+                contentStringBuilder.append(content);
+            } else if (content instanceof JAXBElement<?>) {
+                var element = (JAXBElement<?>) content;
```

Comment:
extract to method with derscriptive name. e.g. converXmlElementFoundInFreeTextToString

Reply:
There should be fresher changes than this. (amongst other this should have been fixed)